### DISCUSSION TITLE
Where do I define my renovate.json default file ?

### DISCUSSION THREAD

**Discussion by azro352** [renovatebot/renovate#24242] [Request Help]
Hi, 
I've set up the [renovate-runner][1] and I've notived the [`renovate.json`][2] file in the project, but as in my project I import the CI file, this isn't the file that is used as default

I've tried changing some configuration but without any changes, so I'd prefer be sure that I'm using the correct file before more debugging on my side

I guess I should put my config file (my custom one) in the private project running the scheduled pipelines and importing the runner's CI

Did i understand well and so, should the steps be the following ? I can make a PR in the runner's project to suggest that

```diff
1. Create a new project to host the runner
2. Configure credentials using CI variables
3. Create a new main pipeline that includes this project's template
- 4. Set up a schedule to run the pipeline regularly
+ 4. Add your custom `renovate.json` in this new project
+ 5. Set up a schedule to run the pipeline regularly
```

> Self-hosted // v36.79.1-slim // gitlab.com

Thanks

[1]: https://gitlab.com/renovate-bot/renovate-runner
[2]: https://gitlab.com/renovate-bot/renovate-runner/-/blob/main/renovate.json?ref_type=heads

**Comment by rarkins** [comment]
No that's not correct.

Custom bot config goes in a config.js or via env variables

**Reply by azro352** [reply]
Ok get it, a config.je in the base dir of the repo including the runner's CI

But how can I define global datasources or global hostRules ? I only see the documentation for them in the renovate.json for the repo config

**Reply by rarkins** [reply]
You can define all of those in a `config.js` file too

**Reply by azro352** [reply]
I can't find how , I tried the following without success it isn't used 

And I don't understand if I have to use packageRules or not 

```js 
module.exports = {
  repositories : [
    {
      repository : "*", 
      hostRules : { matchHost : "https://myrepo", hostType : "pypi"}
    }
  ]
}
```

**Reply by rarkins** [reply]
That's not a valid `config.js` even if it were used. You need something like this:

```
module.exports = {
      hostRules : [{ matchHost : "https://some.registry", hostType : "pypi", token: process.env.PIP_TOKEN }]
}
```

But you can also define such hostRules using `RENOVATE_HOST_RULES` in env instead of needing `config.js`

**Reply by azro352** [reply]
Ok, i didn't understand y that repo specific config could also be global 

But that doesn't work, it still look for my packages on pypi.org, what can I try ?

**Reply by rarkins** [reply]
If you goal was to change the default registry from pypi to your private server then your config was completely wrong - hostRules is not the way. It sounds like you need packagesRules with registryUrls, but you are not describing clearly enough what you want for me to be sure.

**Comment by rarkins** [comment]
The topic now has gone very far from the original question so this thread should be closed and locked.
